### DISCUSSION TITLE
Windows 同步助手无法使用的解决方案（已解决）

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2129] [重要通知]
# 注意

同步助手 1.0.2 版应该解决了多个同步助手同时运行的问题，详细描述 [请看这里](https://github.com/Kenshin/simpread/discussions/2129#discussioncomment-2910726)。

## 问题描述

RT，运行同步助手后，如果出现下图所示内容的话，说明 **7026 端口 被占用**

![image](https://user-images.githubusercontent.com/47232992/117907082-bc962980-b308-11eb-9cd2-93f735172895.png)

## 解决方案

> 下面的解决方案不代表共性，但发生问题是，可以通过此方式排查。

报错的原因是开机启动了两次同步助手，因为第一个同步助手占用了端口，第二个会报错。
![Snipaste_2021-05-12_10-40-06](https://user-images.githubusercontent.com/47232992/117913250-a3df4100-b313-11eb-82bf-d935756f7fc3.png)

## 解决办法

>  在任务管理器清理 windows 的自启项，关闭同步助手中的“随系统启动”选项，然后按照这个https://github.com/Kenshin/simpread/discussions/2011#discussion-3312702 设置自启，保证只用这个启动就没问题了。

![Snipaste_2021-05-12_11-24-45](https://user-images.githubusercontent.com/47232992/117913835-af7f3780-b314-11eb-9b9d-e568750e2395.png)
不知道题主是不是打开了两次同步助手造成的。

以上内容由社区用户 @DavidJoyDong 发现并解决，详细请看 https://github.com/Kenshin/simpread/issues/2105 的说明。

***

## 问题描述

> 点开同步助手的exe文件时报错，简悦版本是2.2，助手版本是1.0.1

![image](https://user-images.githubusercontent.com/81074/120993669-59d46880-c7b6-11eb-92a1-4a737e85b7df.png)

## 解决方案

> 运行 [32位](http://ksria.com/simpread/docs/#/Sync?id=%e4%b8%8b%e8%bd%bd) 版本即可解决。

**Comment by Kenshin** [comment]
## 安装同步助手 64 位版本，会被微软 Defender 报毒

![已阻止威胁](https://user-images.githubusercontent.com/16662089/118914376-b124ad00-b95d-11eb-881a-3e77794bbb19.png)

## 注意

如果是通过 [正规渠道下载](http://ksria.com/simpread/docs/#/Sync?id=%e4%b8%8b%e8%bd%bd) 的话，**请不用理会此问题**。

## 解决方案

1. 试试 32 位系统
2. 加白名单

**Reply by LivingElect** [reply]
thanks

**Comment by mflu** [comment]
问题描述:
启动同步助手, 会报告没有权限监听端口
![image](https://user-images.githubusercontent.com/1452633/159103652-226bd991-760a-494a-9458-00c386717b52.png)
如何解决呢?

**Reply by Kenshin** [reply]
应该是你多开了同步助手造成的端口被占用，重启电脑或者杀掉同步助手的进程。

**Reply by mflu** [reply]
应该不是, 我是重启电脑和杀掉进程都做过了, 依旧服务启动. 报警信息显示的不是端口被占用,而是没有权限

**Reply by Kenshin** [reply]
emmm

是不是有什么防火墙之类的在运行？
我记得 Windows 也有查看端口号是否被占用的命令行，你可以试试。

**Comment by Kenshin** [comment]
> 嗨，各位，新版同步助手来啦。

![welcome-synchelper](https://user-images.githubusercontent.com/81074/172756557-608b3e22-085f-43da-b2c7-d008503db009.png)

# 单例运行

当同时运行多个同步助手时，仅确保只有一个运行成功，就可以解决 7026 端口因多开同步助手而导致的占用问题。

同时也修复了开机无法自启动的问题。

## 下载地址

https://simpread.pro/download

## 更新说明

1. [Telegram Channel](https://t.me/simpread/407)

2. [Notion](https://simpreadpublish.notion.site/simpreadpublish/2482effc334e4df8aa3aac7da8bb8059#15e6f9d3f41f483b8e005ddf3b23e255)

## 如何升级

1. 打开同步助手后会自动提示升级。
2. 通过上面的下载地址手动升级。

# 不止如此

1.0.2 版还增加了以下新功能：

- [PDF 高级打印方式](https://github.com/Kenshin/simpread/discussions/3887)，自带目录结构、提供了定制化服务，包含主题、字体、页边距等细节。

- [日志输出](http://ksria.com/simpread/docs/#/Sync?id=%E6%97%A5%E5%BF%97)。

- 将导出文件导出到任意文件夹的 [增强导出方案](http://ksria.com/simpread/docs/#/Sync?id=%E5%A2%9E%E5%BC%BA%E5%AF%BC%E5%87%BA)。

不止如此，**优化和修复 60+ Isues**，包括：修复自启动无效、隐藏 Dock 栏，Menubar icon 颜色自适应等，墙裂建议用户升级。

# 附录

以下是自定义导出 PDF 的一些截图

![image](https://user-images.githubusercontent.com/81074/172757646-37b4e8ee-19c7-4f99-ba39-3fbdecc50c3e.png)

![image](https://user-images.githubusercontent.com/81074/172757663-f773c9aa-9650-46ec-84e2-c08e868c6244.png)

![image](https://user-images.githubusercontent.com/81074/172757678-7e00bdbe-aefc-4713-9648-55e6ff87993b.png)

# 彩蛋

同步助手 1.0.2 发布后即将进入「快车道」 更新的新版已经在测试了。😄

***

我关闭这个 Issues 了，有任何问题，仍可随时在这里讨论。😄
